api/src/user/user.handlers.ts

```diff
@@ -247,3 +250,167 @@ export const getUserWorksheet = async (
     data: worksheetCoursesToWorksheets(worksheets)[netId] ?? {},
   });
 };
+
+const AddWorksheetSchema = z.object({
+  action: z.literal('add'),
+  season: z.string().transform((val) => parseInt(val, 10)),
+});
+
+const DeleteWorksheetSchema = z.object({
+  action: z.literal('delete'),
+  season: z.string().transform((val) => parseInt(val, 10)),
+  worksheetNumber: z.number(),
+});
+
+const RenameWorksheetSchema = z.object({
+  action: z.literal('rename'),
+  season: z.string().transform((val) => parseInt(val, 10)),
+  worksheetNumber: z.number(),
+  worksheetName: z.string().max(64),
+});
+
+const UpdateWorksheetMetadataSchema = z.union([
+  AddWorksheetSchema,
+  DeleteWorksheetSchema,
+  RenameWorksheetSchema,
+]);
+
+export const updateWorksheetMetadata = async (
+  req: express.Request,
+  res: express.Response,
+): Promise<void> => {
+  winston.info('Updating worksheets metadata');
+
+  const { netId } = req.user!;
+
+  const bodyParseRes = UpdateWorksheetMetadataSchema.safeParse(req.body);
+  if (!bodyParseRes.success) {
+    res.status(400).json({ error: 'INVALID_REQUEST' });
+    return;
+  }
+
+  const { action, season } = bodyParseRes.data;
+
+  if (action === 'add') {
+    winston.info(`Adding worksheet for user ${netId}`);
+
+    const worksheetNumbersRes = await db
+      .select({ worksheetNumber: worksheetMetadata.worksheetNumber })
+      .from(worksheetMetadata)
+      .where(
+        and(
+          eq(worksheetMetadata.netId, netId),
+          eq(worksheetMetadata.season, season),
+        ),
+      )
+      .orderBy(worksheetMetadata.worksheetNumber);
+
+    const worksheetNumbers = worksheetNumbersRes.map(
+      (row) => row.worksheetNumber,
+    );
+
+    const firstAvailableWsNumber = getFirstAvailableWsNumber(worksheetNumbers);
```

Comment:
Don't you need to return this number to client side? Or are you just gonna reload and refetch everything